### DISCUSSION TITLE
Line (or area) chart with partial segments

### DISCUSSION THREAD

**Discussion by RobinBertilsson** [airbnb/visx#1776] [Q&A]
Hi again 👋
What do you think should be the approach to archive the following (see image)?

Would it be possible to use the high-level `XYChart` component, or would you have to dig deeper?
Regarding splitting the line into (in this case) two segments I guess I can use `SplitLinePath`, but would you do the partial area thing?

<img width="901" alt="image" src="https://github.com/airbnb/visx/assets/4040035/760ecd58-9a1f-400a-a45f-b17153ce315c">

I really appreciate any help you can provide.
/R

**Comment by RobinBertilsson** [comment]
Anyone? I could really need some help 😅

**Comment by williaster** [comment]
the easiest way to do this is to split the data into two, one for past/future, and render two series. here's a quick [demo](https://codesandbox.io/p/sandbox/custom-theme-forked-474gyn?file=%2FExample.tsx%3A14%2C20&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clq5we7wv00063b6l9dweme6g%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clq5we7wu00023b6lg4vtua7u%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clq5we7wu00033b6l9tm3qlcl%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clq5we7wu00053b6lkbxq7rhp%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clq5we7wu00023b6lg4vtua7u%2522%253A%257B%2522id%2522%253A%2522clq5we7wu00023b6lg4vtua7u%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clq5wfzlp007i3b6lljp8zj34%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A14%252C%2522startColumn%2522%253A20%252C%2522endLineNumber%2522%253A14%252C%2522endColumn%2522%253A20%257D%255D%252C%2522filepath%2522%253A%2522%252FExample.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clq5wfzlp007i3b6lljp8zj34%2522%257D%252C%2522clq5we7wu00053b6lkbxq7rhp%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clq5we7wu00043b6lxcxk1lv6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clq5we7wu00053b6lkbxq7rhp%2522%252C%2522activeTabId%2522%253A%2522clq5we7wu00043b6lxcxk1lv6%2522%257D%252C%2522clq5we7wu00033b6l9tm3qlcl%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clq5we7wu00033b6l9tm3qlcl%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) 

<img width="1233" alt="image" src="https://github.com/airbnb/visx/assets/4496521/6caf1b03-802f-440e-9477-3ffc916eea13">

BUT  where this gets hard is the interpolation. if you have `curveLinear` (ie no interpolation) it looks fine, but if you have curvy lines like your design, the interpolation on the end points of the two series won't match and (depending on the data shape) it will look bad. that's why `splitlinepath` exists. it currently doesn't work with `xychart` but it could be integrated or you could write a custom series component that uses it under the hood.

<img width="1226" alt="image" src="https://github.com/airbnb/visx/assets/4496521/439a1dfd-9a6e-4706-b692-7aaadc74a928">

**Reply by RobinBertilsson** [reply]
Thanks for your help! I actually got it working just prior to your answer, but as you said regarding the curve, the `<SplitLinePath />` was very much needed, and solved it all once I got the hang of it.

I'll make sure to share my code here, for others that might be facing the same issue.

**Reply by imjared** [reply]
@RobinBertilsson before we get too far into the future, wondering if you might still have your code available

**Reply by RobinBertilsson** [reply]
I'm afraid i dont, but i'm quite sure it was just a matter of splitting the data into multiple segments, e.g. `[1, 2, 3, 4]` -> `[[1, 2], [3, 4]]` and then use `<SplitLinePath />`, it was much easier than i thought.
